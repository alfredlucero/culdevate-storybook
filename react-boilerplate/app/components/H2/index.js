import styled from 'styled-components';

const H2 = styled.h2`
  color: ${({ primary }) => primary ? '#58a1e3' : '#595959'};
  font-family: 'Inconsolata';
  font-size: 1.5em;
`;

export default H2;
