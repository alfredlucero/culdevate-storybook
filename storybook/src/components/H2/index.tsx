import * as React from 'react';

import styled from 'styled-components';

export interface H2Props {
  primary?: boolean;
}

const H2 = styled.h2`
  color: ${(props:H2Props) => props.primary ? '#58a1e3' : '#595959'};
  font-size: 1.5rem;
  font-family: 'Inconsolata';
  letter-spacing: 1px;
`;

export default H2;