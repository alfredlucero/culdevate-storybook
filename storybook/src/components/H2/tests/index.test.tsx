import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import H2 from '../index';

describe('<H2 />', () => {
  const defaultTextColor = '#595959';
  const primaryTextColor = '#58a1e3';

  it('should render correctly', () => {
    const tree = renderer.create(<H2 />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render default color H2 given no props', () => {
    const tree = renderer.create(<H2 />).toJSON();
    expect(tree).toHaveStyleRule('color', defaultTextColor);
  });

  it('should render primary color H2 given primary prop', () => {
    const tree = renderer.create(<H2 primary />).toJSON();
    expect(tree).toHaveStyleRule('color', primaryTextColor);
  });

  it('should render its text', () => {
    const children = 'Sample H2 Text';
    const renderedComponent = shallow(
      <H2>{children}</H2>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });
});

