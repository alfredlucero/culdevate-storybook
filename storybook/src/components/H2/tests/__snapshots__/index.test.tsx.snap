// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> should render correctly 1`] = `
.c0 {
  color: #595959;
  font-size: 1.5rem;
  font-family: 'Inconsolata';
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

<h2
  className="c0"
/>
`;
